src/envoy/utils/utils.h

```diff
@@ -41,6 +41,10 @@ bool GetDestinationUID(const envoy::api::v2::core::Metadata& metadata,
 bool GetPrincipal(const Network::Connection* connection, bool peer,
                   std::string* principal);
 
+// Get source.namespace attribute from principal.
+bool GetSourceNamespace(const std::string& principal,
```

Comment:
Does missing namespace is a valid principal?

Reply:
I think all principal would have the namespace?

Reply:
What I mean is if it is 'required', then should return false should be treated as authentication fail? In the same way that principal is not in the correct format.

Reply:
I don't know if it's required so I feel it's a bit dangerous to treat this as authentication fail though from the code in citadel, it seems it should always be there.
Also even if we should treat it as a fail, I will do it in a different smaller PR which would change the current behavior.